### DISCUSSION TITLE
Instructions unclear - what am I building?

### DISCUSSION THREAD

**Discussion by wicol** [DevSkillsHQ/discusssions#33] [Help]
Hey, trying to do this assignment called "Interactive Presentation Polls & Votes Backend - Level 2".

The readme states I should proxy the requests related to presentations to `https://infra.devskills.app/api/interactive-presentation/v4`. And that I only have to store Polls and Votes in my database. However, looking at the endpoints I'm proxying to, the client is expected to supply the polls when creating a presentation. The polls and the `current_poll_index` are then available to fetch from that API as well. Meanwhile I'm supposed to store the polls and information about which poll is the current in my DB? I'm confused. Should I ignore some data in that API and work with my own? Can I just implement the Presentation model in my DB instead..? Is this all part of the test?? :D

Btw the tests don't run as stated in your docker example because of the ancient version tag.

**Comment by fenske** [comment]
Hi @wicol. I'm Anton from Alva. Will do my best to clarify the situation here.

The server at `https://infra.devskills.app/api/interactive-presentation/v4` indeed implements all endpoints from the aforementioned API. You're asked to only proxy some of the responses because the main objective of this particular task is to assess how well the candidate can deal with modeling a DB and work with third-party APIs. So that's the reason why you're asked to create presentations on the remote server while tracking the current poll position for a particular presentation (which details you can fetch from the remote server) & handling votes in the DB model of your app.

Good catch with the Docker part. We've just upgraded Cypress to v12, so the tag should update as well. In the meantime, you can either use npm or GA pipeline to run the tests (if you don't want to install NPM, you can spin up a Gitpod using the instructions in the README).

Hope this clarifies it. Let me know if you have any other questions.

**Comment by wicol** [comment]
Sure, I get that you want both DB handling and interaction with other APIs. My confusion was around the statement "This means that you'll only need to store polls and votes in your database.". That doesn't seem right, since I don't actually need to store polls in order to fullfill the requirements, but I do need to store current indexes for presentations (which are also stored in the 3rd party API but cannot be modified). Anyways, closing this and getting on with it.
